### DISCUSSION TITLE
Rule proposal: No Async Lifecycles

### DISCUSSION THREAD

**Discussion by DavidANeil** [angular-eslint/angular-eslint#1078] [New Rule Proposals]
Lifecycle methods (e.g. `ngOnInit`, `ngOnDestroy`) are called synchronously by Angular, so async methods (or really any return value) is ignored.
It would be nice to be able to enforce that these methods are not `async`, or more generally that they do not return a `Promise`, or potentially even any return value could be considered an error.

This would help catch issues that can arise in testing components where the `ngOnDestroy` is called during cleanup, but the asynchronous tasks it spawns might execute during a different test.

**Comment by chimurai** [comment]
related issue https://github.com/angular-eslint/angular-eslint/issues/1515
implemented in https://github.com/angular-eslint/angular-eslint/pull/1643

`@angular-eslint/no-async-lifecycle-method` rule released in [v17.2.0](https://github.com/angular-eslint/angular-eslint/releases/tag/v17.2.0)
